### PR TITLE
CORTX-29196: Add more capabilities to the admin user.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
Signed-off-by: Shriya Deshmukh <shriya.deshmukh@seagate.com>

# Problem Statement
- Add more capabilities to the admin user.
**"user", "users", "buckets", "metadata", "info", "usage", "zone", "roles", "user-policy"**

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by shriya-deshmukh** [comment]
radosgw-admin cmd : /usr/bin/radosgw-admin user create --uid=test_user --display-name=test_user  -c /etc/cortx/rgw/e884506cadb150157e923ad8e4873ac4/cortx_rgw.conf -n client.radosgw-admin --caps="user=*;users=*;metadata=*;usage=*;zone=*;roles=*;user-policy=*;buckets=*;info=*" --no-mon-config
output:
```
{
    "user_id": "test_user",
    "display_name": "test_user",
    "email": "",
    "suspended": 0,
    "max_buckets": 1000,
    "subusers": [],
    "keys": [
        {
            "user": "test_user",
            "access_key": "K2HZBKW64ROH3JRKGHIA",
            "secret_key": "K2A8EKrv7EkCi1qLgTPJtcYa8FM6xe1eMi5ykH2j"
        }
    ],
    "swift_keys": [],
    "caps": [
        {
            "type": "buckets",
            "perm": "*"
        },
        {
            "type": "info",
            "perm": "*"
        },
        {
            "type": "metadata",
            "perm": "*"
        },
        {
            "type": "roles",
            "perm": "*"
        },
        {
            "type": "usage",
            "perm": "*"
        },
        {
            "type": "user",
            "perm": "*"
        },
        {
            "type": "user-policy",
            "perm": "*"
        },
        {
            "type": "users",
            "perm": "*"
        },
        {
            "type": "zone",
            "perm": "*"
        }
    ],
    "op_mask": "read, write, delete",
    "default_placement": "",
    "default_storage_class": "",
    "placement_tags": [],
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "user_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "temp_url_keys": [],
    "type": "rgw",
    "mfa_ids": []
}
```

**Review by sachitanands — APPROVED** [review]
looks good

**Review by sachinpunadikar — APPROVED** [review]
+1

File: src/rgw/setup/rgw.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def _create_rgw_user(conf: MappedConf):
         rgw_config = Rgw._get_rgw_config_path(conf)
         create_usr_cmd = f'radosgw-admin user create --uid={user_name} --access-key \
             {access_key} --secret {password} --display-name="{user_name}" \
-            --caps="users=*;metadata=*;usage=*;zone=*" \
+            --caps="users=*;metadata=*;usage=*;zone=*;info=*;user=*;roles=*;user-policy=*;buckets=*" \
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:277]: Enabling all permissions blindly can raise some security concerns later, we should add specific roles & capabilities and not wildcard * i believe.
@sachinpunadikar aren't there any specific values we should be adding ?

Comment (sachinpunadikar) [reply src/rgw/setup/rgw.py:277]: For all capabilities there are only 2 values, READ & WRITE like below:
#define RGW_CAP_READ            0x1
#define RGW_CAP_WRITE           0x2
#define RGW_CAP_ALL             (RGW_CAP_READ | RGW_CAP_WRITE)
So here * means both capabilities. To get the values we need READ & and for set/modify, we need WRITE capability. So current capabilities to admin user are fine, no need to change.

Comment (cdeshmukh) [reply src/rgw/setup/rgw.py:277]: Thanks @sachinpunadikar for confirming. @shriya-deshmukh old me on simlar lines yday, still wanted to double confirm

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**